dashboard/app/models/pd/enrollment.rb

```diff
@@ -176,19 +176,23 @@ def resolve_user
   end
 
   def exit_survey_url
-    if [Pd::Workshop::COURSE_ADMIN, Pd::Workshop::COURSE_COUNSELOR].include? workshop.course
-      CDO.code_org_url "/pd-workshop-survey/counselor-admin/#{code}", CDO.default_scheme
-    elsif workshop.summer?
-      pd_new_workshop_survey_url(code, protocol: CDO.default_scheme)
-    elsif [Pd::Workshop::COURSE_CSP, Pd::Workshop::COURSE_CSD].include?(workshop.course) && workshop.workshop_starting_date > Date.new(2018, 8, 1)
-      CDO.studio_url "/pd/workshop_survey/day/#{workshop.sessions.size}?enrollmentCode=#{code}", CDO.default_scheme
-    else
-      CDO.code_org_url "/pd-workshop-survey/#{code}", CDO.default_scheme
-    end
+    return CDO.code_org_url "/pd-workshop-survey/counselor-admin/#{code}", CDO.default_scheme if
```

Comment:
I use guard clause like this instead of `if-elsif-elsif...` to make it less error-prone and easier to read. [Reference](https://medium.com/the-code-review/en-garde-how-you-can-refactor-nested-ifs-with-guard-clauses-in-ruby-1ff728b7a0cd). Please let me know which style you prefer?

Reply:
I prefer the original `if-elsif-else` structure of this method.

I'm not against guard clauses as a pattern. I just don't think this is the right place to use them.  The article you reference is all about avoiding _nested_ if/else logic for readability, but there's no nesting here.  It's more of a one-layer switch statement.

More specifically, I find guard clauses useful for validating input or handling unexpected cases. For example:

```rb
def my_workshops
  return [] unless current_user
  # Happy-path logic
end
```

My interpretation of this method is that all of the cases are happy-path cases, likely to occur in ordinary use.  For that reason, I think the `if-elsif-else` structure is a clearer expression of the intended behavior of the method.  I also think the elegance of the method as a single expression is very ruby-idiomatic.

I understand that's very touchy-feely-squishy code style reasoning, and a reasonable person might come to the opposite conclusion.  Therefore I'm interested in other opinions.  @clareconstantine and @Hamms do you have thoughts on this?

Reply:
Thanks guys for the input. I'm okay with keeping it as-is now that I understand what it does. (At first it was quite confusing reading 9 condensed lines with multiple conditions).

Reply:
I could really go either way on this; the things that make this method hard to reason about for me are the opaqueness of the conditions themselves and the repeated code in the url generation.

Would y'all feel differently about guard clauses vs if/else if the method looked like this?

```ruby
def exit_survey_url
  return pd_counselor_admin_workshop_survey_url(code) if workshop.counselor_admin?
  return pd_new_workshop_survey_url(code) if workshop.summer?
  return pd_csd_csp_2018_workshop_survey_url(workshop.sessions.size, code) if workshop.csd_csp_2018?
  return pd_csf_2018_workshop_survey_url(code) if workshop.csf_201?

  pd_workshop_survey_url(code)
end
```

Reply:
I agree that reads better than either approach given the current conditions, but I still feel a slight mismatch using guard-clauses for something more like a case statement (this might be the single-return-rule hammered into me as an undergrad rearing its ugly head).  In fact, I might prefer an actual case statement, with the conditions on the left:

```rb
def exit_survey_url
  case
  when workshop.counselor_admin? then pd_counselor_admin_workshop_survey_url(code)
  when workshop.summer?          then pd_new_workshop_survey_url(code)
  when workshop.csd_csp_2018?    then pd_csd_csp_2018_workshop_survey_url(workshop.sessions.size, code)
  when workshop.csf_201?         then pd_csf_2018_workshop_survey_url(code)
  else                                pd_workshop_survey_url(code)
  end
end
```

Reply:
Unrelated: After staring at this method for a while, it's starting to feel like it should live on `Workshop` instead of `Enrollment`.

Reply:
Reverting this back to use `if-elsif-else`. 2 open questions that we might want to address later:
1. Is there a better way to write multiple branches with long, sometime complex, conditions. I know the pain of reading it will fade away once you understand it, however that feeling will come back when you forget what it does  :)
2. Survey link(s) should be in Pd::Workshop rather than Pd::Enrollment.